sdk/python/kfp/compiler/test_data/pipelines/pipeline_with_nested_exit_handlers.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from kfp import compiler
+from kfp import dsl
+
+
+@dsl.component
+def producer_op() -> str:
+    return 'a'
+
+
+@dsl.component
+def dummy_op(msg: str = '') -> str:
+    return msg
+
+
+@dsl.pipeline(name='pipeline-with-nested-exit-handlers')
```

Comment:
I would avoid this nested example unless 1) we have explicit ask for this feature, and 2) we confirmed there's a support path (via Argo) on KFP. 
Before we officially support this feature, we should ideally block it from compiling.

Reply:
That makes sense to me. I left the logic, but added a compilation guard and a comment in debfe4b.